### PR TITLE
Fixed #15779 -- Fixed UserChangeForm regression introduced by 1791a7e75

### PR DESCRIPTION
**Pull request by claudep**
Thanks Tim Graham for reporting the regression.

### CODE REVIEW

File: tests/auth_tests/test_views.py [hunk 1]
```diff
@@ -943,14 +944,27 @@ def test_user_not_change(self):
...
+        user_change_url = reverse('auth_test_admin:auth_user_change', args=(self.admin.pk,))
+        password_change_url = reverse('auth_test_admin:auth_user_password_change', args=(self.admin.pk,))
+
+        response = self.client.get(user_change_url)
+        # Test the link inside password field help_text.
+        rel_link = re.search(
+            r'you can change the password using <a href="([^"]*)">this form</a>', response.content
+        ).groups()[0]
+        self.assertEqual(
+            os.path.normpath(user_change_url + rel_link),
```
Comment (MarkusH) [inline tests/auth_tests/test_views.py]: Does Windows use `/` or `\` in normalized paths? In case of `\`, does the test still pass?

Comment (claudep) [inline tests/auth_tests/test_views.py]: You are right, on Windows, slashes are converted into backslashes. However, the test should still pass as normpath is called both sides. Not tested though.

File: django/contrib/auth/forms.py [hunk 2]
```diff
@@ -98,7 +98,7 @@ class UserChangeForm(forms.ModelForm):
     password = ReadOnlyPasswordHashField(label=_("Password"),
         help_text=_("Raw passwords are not stored, so there is no way to see "
                     "this user's password, but you can change the password "
-                    "using <a href=\"password/\">this form</a>."))
+                    "using <a href=\"../password/\">this form</a>."))
```
Comment (charettes) [inline django/contrib/auth/forms.py]: Would the following considered over-engineered?

```python
from django.core.urlresolvers import reverse
from django.utils.functional import lazy
from django.utils import six
from django.utils.translation import ugettext

def _password_helptext():
    password_change_url = reverse('password_change')
    return ugettext('...using <a href="%(password_change_url)s">this form</a>') % {'password_change_url': password_change_url}
password_helptext = lazy(_password_helptext, six.text_type)
```

Comment (claudep) [inline django/contrib/auth/forms.py]: The problem is that we are not talking about `password_change`, but `auth_user_password_change` which takes the user pk as argument, and that complicates things a bit more.

Comment (charettes) [inline django/contrib/auth/forms.py]: Ahh I see, didn't realize that.

**Comment by timgraham** [comment]
LGTM. When I fix a follow-up regression, I usually use "Refs" instead of "Fixed" in the commit message to avoid confusing the commit with the main "fix" for the ticket.

**Pull request merged**